pkg/core/resources/model/resource.go

```diff
@@ -409,12 +409,19 @@ func IsReferenced(refMeta ResourceMeta, refName string, resourceMeta ResourceMet
 }
 
 func IsLocallyOriginated(mode config_core.CpMode, r Resource) bool {
+	if r.GetMeta() == nil {
```

Comment:
we are checking this because lots of tests panicked without this check.

It feels wrong to return `false` in this case, but when I started from `true` the whole functionality didn't work and we were still setting namespace labels on synced resources. I checked it now and defaulter is running on every store operation as this is kubernetes webhook. So probably we don't need to add namespace in store, it would be enough to update defaulter

Reply:
It was probably panic when called from the Create, because ResourceMeta is not set. If you call this only from defaulter now, maybe we don't need this `nil` check

Reply:
removed check